pkg/lib/serverTime.js

```diff
@@ -321,13 +324,181 @@ export function ServerTime() {
                     console.warn("failed to load time servers", error);
                     return result;
                 });
-    };
+    }
 
-    self.set_custom_ntp = function (servers, enabled) {
-        const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${enabled ? "" : "#"}NTP=${servers.join(" ")}\n`;
+    function set_ntp_config_timesyncd(config) {
+        const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
+                                                    { superuser: "try" });
+
+        const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${config.enabled ? "" : "#"}NTP=${config.servers.join(" ")}\n`;
 
         return cockpit.spawn(["mkdir", "-p", "/etc/systemd/timesyncd.conf.d"], { superuser: "try" })
                 .then(() => custom_ntp_config_file.replace(text));
+    }
+
+    function find_file_and_read(names, first) {
+        if (first === undefined)
+            first = 0;
+        if (first >= names.length)
+            return Promise.reject(new Error("Found none of " + names.join(",")));
+        return cockpit.file(names[first], { superuser: "try" }).read()
+                .then(text => {
+                    if (text)
+                        return [names[first], text];
+                    return find_file_and_read(names, first + 1);
+                });
+    }
+
+    function get_ntp_config_chronyd() {
+        return find_file_and_read(["/etc/chrony.conf", "/etc/chrony/chrony.conf"])
```

Comment:
Note to self: RHEL/Fedora uses the former, Debian's entrypoint is the latter. Debian also has has a pair of standrad /etc/chrony/conf.d/* and /etc/chrony/sources.d/* drop-in directories which packages or admins are supposed to customize stuff in.

So in that regard, *if* chrony.conf has an existing `confdir` line already, then it would make sense for cockpit to write a conf.d/cockpit.conf snippet, then we can avoid modifying the main config file completely.